nomenclature/cli.py

```diff
@@ -269,3 +271,118 @@ def validate_scenarios(
         If input_file validation fails against specified codelist(s).
     """
     DataStructureDefinition(definitions, dimensions).validate(IamDataFrame(input_file))
+
+
+@app.command()
+def convert_xlsx_codelist_to_yaml(
+    source: Annotated[Path, typer.Argument(..., exists=True)],
+    target: Annotated[Path, typer.Argument(...)],
+    sheet_name: Annotated[str, typer.Argument(...)],
+    col: Annotated[str, typer.Argument(...)],
+    attrs: Annotated[list[str] | None, typer.Option()] = None,
+):
+    """Parses an xlsx file with a codelist and writes a yaml file
+
+    Parameters
+    ----------
+    source : str, path, file-like object
+        Path to xlsx file with definitions (codelists).
+    target : str, path, file-like object
+        Path to save the parsed definitions as yaml file.
+    sheet_name : str
+        Sheet name of `source`.
+    col : str
+        Column from `sheet_name` to use as codes.
+    attrs : list, optional
+        Columns from `sheet_name` to use as attributes.
+    """
+    if attrs is None:
+        attrs = []
+    SPECIAL_CODELIST.get(col.lower(), CodeList).read_excel(
+        name="", source=source, sheet_name=sheet_name, col=col, attrs=attrs
+    ).to_yaml(target)
+
+
+@app.command()
+def parse_model_registration(
+    model_registration_file: Annotated[Path, typer.Argument(..., exists=True)],
+    region_path: Annotated[Path, typer.Option(exists=True)] = (
+        Path.cwd() / "definitions" / "region"
+    ),
+    mappings_path: Annotated[Path, typer.Option(exists=True)] = Path.cwd() / "mappings",
+) -> None:
+    """Parses a model registration file and writes the definitions & mapping yaml files
+
+    Parameters
+    ----------
+    model_registration_file : path, file-like object
+        Path to xlsx model registration file.
+    region_path : path
```

Comment:
Suggest to rename for clarity.
```suggestion
    definition_path : path
```